apis/workstations/v1alpha1/workstationsworkstation_types.go

```diff
@@ -0,0 +1,138 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var WorkstationsWorkstationGVK = GroupVersion.WithKind("WorkstationsWorkstation")
+
+// WorkstationsWorkstationSpec defines the desired state of WorkstationsWorkstation
+// +kcc:proto=google.cloud.workstations.v1.Workstation
+type WorkstationsWorkstationSpec struct {
+	// The WorkstationsWorkstation name. If not given, the metadata.name will be used.
+	// + optional
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	// // Full name of this workstation.
+	// Name *string `json:"name,omitempty"`
+
+	// Optional. Human-readable name for this workstation.
+	DisplayName *string `json:"displayName,omitempty"`
+
+	// Optional. Client-specified annotations.
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// NOTYET: Not dealing with labels yet
+	// // Optional.
+	// //  [Labels](https://cloud.google.com/workstations/docs/label-resources) that
+	// //  are applied to the workstation and that are also propagated to the
+	// //  underlying Compute Engine resources.
+	// Labels map[string]string `json:"labels,omitempty"`
+
+}
+
+// WorkstationsWorkstationStatus defines the config connector machine state of WorkstationsWorkstation
+type WorkstationsWorkstationStatus struct {
+	/* Conditions represent the latest available observations of the
+	   object's current state. */
+	Conditions []v1alpha1.Condition `json:"conditions,omitempty"`
+
+	// ObservedGeneration is the generation of the resource that was most recently observed by the Config Connector controller. If this is equal to metadata.generation, then that means that the current reported status reflects the most recent desired state of the resource.
+	// +optional
+	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+
+	// A unique specifier for the WorkstationsWorkstation resource in GCP.
+	// +optional
+	ExternalRef *string `json:"externalRef,omitempty"`
+
+	// ObservedState is the state of the resource as most recently observed in GCP.
+	// +optional
+	ObservedState *WorkstationsWorkstationObservedState `json:"observedState,omitempty"`
+}
+
+// WorkstationsWorkstationSpec defines the desired state of WorkstationsWorkstation
+// +kcc:proto=google.cloud.workstations.v1.Workstation
+type WorkstationsWorkstationObservedState struct {
```

Comment:
I saw the recent Direct resources all have the GCP `observedState` fields added in `.status`. This may be something worth discussing. I see the value of adding those fields (and maybe more than not adding them), though 
my concerns are: 
1. We may then need to update the KRM object more frequently without a clear reason. i.e. some output-only fields are changed, like `UpdateTime`. 
2. It makes the "snapshotting" harder. Snapshot refers to the ConfigConnector's approach to hash previous state changes, and determine whether to update/reconcile the GCP object or not, like etag.
3. Some observedState field may be too large, like the MonitoringDashboard that can hit size issues more quickly.